doc/specs/stdlib_hash_functions.md

```diff
@@ -0,0 +1,1697 @@
+Some of the hash functions have alternatives that allow incremental
+hashing. 
+
+|Algorithm|Seed|Result|
+|---------|----|------|
+|FNV-1|None|32 or 64 bit integer|
+|FNV-1a|None|32 or 64 bit integer|
+|nmhash32 |32 bit scalar integer|32 bit integer|
+|nmhash32x |32 bit scalar integer|32 bit integer|
+|pengyhash |32 bit scalar integer|64 bit integer|
+|Spooky Hash|64 bit two element vector|64 bit two element vector|
+|waterhash|64 bit scalar integer|32 bit integer|
+
+The hash function modules each provide at least five algorithms for
+hash functions: two optimized for small (< 32 `INT8` integer elements)
+keys, and three optimized for large (> 100 `INT8` integer elements)
+keys.
+The core implementation for each algorithm is for keys that are
+vectors of `INT8` integers.
+These core implementations are then used in wrappers for keys
+that are vectors of `INT16`, `INT32` and `INT64` integers, or default
+character strings, in the expectation that inlining will eliminate the
+overhead of transferring the other keys to `INT8` integer vectors.
+
+The `stdlib_32_bit_hash_functions` module provides
+implementations of five hash code algorithms:
+the *FNV_1* and *FNV_1A* variants of Glenn Fowler, 
+Landon Curt Noll, and Kiem-Phong Vo;
+the *nmhash32* and *nmhash32x* of James Z. M. Gao;
+and the *waterhash*  of Tommy Ettinger.
+The detailed implementation of each algorithm is handled in a separate
+submodule: `stdlib_32_bit_fnv_hashes`,
+`stdlib_32_bit_nmhashes`, and `stdlib_32_bit_water_hashes`,
+respectively. The `nmhash32`, `nmhash32x`, and `waterhash` algorithms
+require seeds. The submodules provide separate seed generators
+for each algorithm.
+The module itself
+implements two scalar hash functions, `FIBONACCI_HASH` and
+`UNIVERSAL_MULT_HASH`. 
+It also implements the subroutine, `ODD_RANDOM_INTEGER`, for
+generating seeds for `UNIVERSAL_MULT_HASH`.
+All assume a two's complement sign bit, and no out of
+range checks.
+
+The `stdlib_64_bit_hash_functions` module also provides
+implementations of four hash code algorithms:
+the *FNV_1* and *FNV_1A* variants of Glenn Fowler, 
+Landon Curt Noll, and Kiem-Phong Vo;
+the *pengynash* of Alberto Fajardo;
+and the *SpookyHash*  of Bob Jenkins.
+The detailed implementation of each algorithm is handled in a separate
+submodule: `stdlib_64_bit_fnv_hashes`,
+`stdlib_64_bit_pengy_hashes`, and `stdlib_64_bit_spooky_hashes`,
+respectively.
+The `pengyhash`, and `Spooky Hash` algorithms
+require seeds. The submodules provide separate seed generators
+for each algorithm.
+The module itself implements two scalar hash functions,
+`FIBONACCI_HASH` and `UNIVERSAL_MULT_HASH`.
+It also implements the subroutine, `ODD_RANDOM_INTEGER`, for
+generating seeds for `UNIVERSAL_MULT_HASH`.
+All assume a two's complement sign bit, and no out of
+range checks.
+
+The `stdlib_32_bit_fnv_hashes` and `stdlib_64_bits_fnv_hashes`
+submodules each provide implementations of ths FNV-1  and FNV-1A
```

Comment:
```suggestion
submodules each provide implementations of ths FNV-1  and FNV-1A
```
```suggestion
submodules each provide implementations of the FNV-1  and FNV-1A
```

Reply:
Changed in my local copy. I also changed the double space after the `FNV-1` to a single space.